### ISSUE TITLE
Decoding to 24bit field returns zero

### ISSUE THREAD

**Issue by arkanoid87** [sealmove/binarylang#15]
I've found a bug while trying to parse a 24 bit field.

```nim
import std/[sugar, strutils]
import binarylang, bitstreams

struct(parser1):
    u23: payload
    u9: filler
dump parser1.get(newStringBitStream("12345678".parseHexStr)).payload.toHex # 00091A2B OK

struct(parser2):
    u25: payload
    u7: filler
dump parser2.get(newStringBitStream("12345678".parseHexStr)).payload.toHex # 002468AC OK

struct(parser3):
    u24: payload
    u8: filler
dump parser3.get(newStringBitStream("12345678".parseHexStr)).payload.toHex # 00000000 NOT OK

struct(parser4):
    u56: payload
    u8: filler
dump parser4.get(newStringBitStream("1234567812345678".parseHexStr)).payload.toHex # 0000563412785634 OK
```

**Comment by arkanoid87** [comment]
ctually not related to 24 bits only, but happents to me in different corner cases. Am I doing something wrong?

```nim
import std/[sugar, strutils]
import binarylang, bitstreams

struct(parser):
    u1: abit
    u31: filler

let data = 0x12345678.int32
dump data
dump data.toHex
dump data.toHex.parseHexStr.len
dump parser.get(newStringBitStream(data.toHex.parseHexStr)).abit

# OUTPUT
# data = 305419896
# data.toHex = 12345678
# data.toHex.parseHexStr.len = 4
# parser.get(newStringBitStream(data.toHex.parseHexStr)).abit = 0
```

**Comment by sealmove** [comment]
This is a bug in `bitstreams`. Specifically `proc readBitsBe`. Thanks for reporting!

**Comment by sealmove** [comment]
Fixed by sealmove/bitstreams@e346580b72616d96ac62e6d9c11d53216ea2bc0f.
